examples/platform/silabs/MatterConfig.cpp

```diff
@@ -294,7 +297,13 @@ CHIP_ERROR SilabsMatterConfig::InitWiFi(void)
 // ================================================================================
 // FreeRTOS Callbacks
 // ================================================================================
+extern "C" uint8_t test;
 extern "C" void vApplicationIdleHook(void)
 {
-    // FreeRTOS Idle callback
+#if ((SIWX_917 == 1) && (SL_ICD_ENABLED == 1))
+    if(ConnectivityMgr().IsWiFiStationConnected()) {
```

Comment:
this doesn't check commissioning. This only checks if we are connected to a wifi network.

Reply:
Actually, the device can go to sleep once it is connected to router and TA will handle the wakeup/sleep